sync-core/src/main/java/com/cloudant/sync/replication/PullReplication.java

```diff
@@ -54,6 +60,66 @@ public PullReplication() {
         /* Does nothing but we can now document it */
     }
 
+    /**
+     * Adds request filters to run for every request made by this replication
+     * @param requestFilters The filters to run
+     * @return The current instance of PullReplication
+     */
+    public PullReplication addRequestFilters(List<HttpConnectionRequestFilter> requestFilters){
```

Comment:
Can these go in the base class, `Replication`? They repeat the same code for Push and Pull